### PR TITLE
Tweak the semantic convention code generation.

### PR DESCRIPTION
**Pull request by jkwatson**
In particular, to better support links and html in the docs.

### CODE REVIEW

**Review by kubawach — APPROVED** [review]
👍

File: buildscripts/semantic-convention/templates/SemanticAttributes.java.j2 [hunk 1]
```diff
@@ -64,10 +64,10 @@ public final class {{class}} {
   {%- for attribute in attributes if attribute.is_local and not attribute.ref %}
 
   /**
-  * {% filter escape %}{{attribute.brief | to_doc_brief}}.{% endfilter %}
+  * {% filter safe %}{{attribute.brief | render_markdown}}{% endfilter %}
```
Comment (Oberon00) [inline buildscripts/semantic-convention/templates/SemanticAttributes.java.j2:67]:
```suggestion
  * {{attribute.brief | render_markdown | safe}}
```

File: buildscripts/semantic-convention/templates/SemanticAttributes.java.j2 [hunk 2]
```diff
@@ -64,10 +64,10 @@ public final class {{class}} {
...
     {%- if attribute.note %}
   *
-  * Note: {% filter escape %}{{attribute.note | to_doc_brief}}.{% endfilter %}
+  * <p>Notes: {% filter safe %}{{attribute.note | render_markdown}}{% endfilter %}
```
Comment (Oberon00) [inline buildscripts/semantic-convention/templates/SemanticAttributes.java.j2:70]:
```suggestion
  * <p>Notes: {{attribute.note | render_markdown | safe}}
```

**Review by Oberon00 — APPROVED** [review]
You could also try using `render_markdown(code="{{@code {0}}}")`. Pro: Will look a bit more idiomatic, con: Would probably break if there are `}` in the input.

**Comment by Oberon00** [comment]
Unfortunately, there is no documentation for this one. 😬
* This is the parameter: https://github.com/open-telemetry/build-tools/blob/v0.4.1/semantic-conventions/src/opentelemetry/semconv/templating/code.py#L42
* Which is used here: https://github.com/open-telemetry/build-tools/blob/v0.4.1/semantic-conventions/src/opentelemetry/semconv/templating/code.py#L102
* This uses Python's `str.format` which uses the syntax documented here: https://docs.python.org/3/library/string.html#formatstrings

**Comment by Oberon00** [comment]
So instead of `<code>sometext</code>` you will end up with `{@code sometext}`.

File: buildscripts/semantic-convention/templates/SemanticAttributes.java.j2 (hunk 1 above)
Comment (jkwatson) [reply buildscripts/semantic-convention/templates/SemanticAttributes.java.j2:67]: I don't even think it needs the `| safe` which I believe is a no-op.

**Comment by jkwatson** [comment]
> Unfortunately, there is no documentation for this one. 😬
> 
> * This is the parameter: https://github.com/open-telemetry/build-tools/blob/v0.4.1/semantic-conventions/src/opentelemetry/semconv/templating/code.py#L42
> * Which is used here: https://github.com/open-telemetry/build-tools/blob/v0.4.1/semantic-conventions/src/opentelemetry/semconv/templating/code.py#L102
> * This uses Python's `str.format` which uses the syntax documented here: https://docs.python.org/3/library/string.html#formatstrings

Also unfortunately, it doesn't look like it works. :(

```
  File "/templates/SemanticAttributes.java.j2", line 67, in template
    * {{attribute.brief | render_markdown(code="{{@code {0}}}"}}
jinja2.exceptions.TemplateSyntaxError: unexpected '}', expected ')'
```

**Comment by Oberon00** [comment]
> Also unfortunately, it doesn't look like it works. :(

Because you forgot the closing `)` between the `"` and the final `}}`. 😃

**Comment by jkwatson** [comment]
> > Also unfortunately, it doesn't look like it works. :(
> 
> 
> 
> Because you forgot the closing `)` between the `"` and the final `}}`. 😃 

Dangit. This code ends up looking like perl.

**Comment by Oberon00** [comment]
Yeah, too many nesting levels here with quotes and various brackets.

**Comment by jkwatson** [comment]
Now I have the power! https://github.com/open-telemetry/opentelemetry-java/pull/3398

**Pull request merged**